### ISSUE TITLE
D417 only triggers when section for arguments is defined

### ISSUE THREAD

**Issue by Mutantpenguin** [PyCQA/pydocstyle#626]
D417 "Missing argument descriptions in the docstring" only triggers when there is a section for arguments, but not when there is none.

If this behaviour is explicitely desired, the documentation should be specific about it.

But my team and myself think that a section for arguments should be enforced if D417 is enabled.

## Examples

This does **not** trigger D417:
```python
def my_function(my_arg: str) -> str:
	"""
	This does something.
	"""

	return my_arg
```

This does trigger D417:
```python
def my_function(my_arg: str) -> str:
	"""
	This does something.

	Arguments:
	"""

	return my_arg
```

## Version
6.3.0

## Config
We use this:
```
convention=google
add-select = D204,D205,D206,D414,D416,D410
add-ignore = D212,D200,D202,D100,D101
```

But the same happens with just:
```
add-select = D417
```
